Dockerfile.jenkins

```diff
@@ -1,16 +1,36 @@
-# includes nodejs + npm
-FROM node:12-alpine
-
+# build stage
+FROM node:12-alpine as build-stage
 RUN apk add --no-cache git
 RUN apk add --no-cache bash
+
 RUN mkdir /ui
 COPY ./ /ui/
 WORKDIR /ui
-# we should switch this to npm ci (need to figure out how to tell Jenkins to maintain some cache of modules first)
+
 RUN npm i --no-save
 RUN npm run build
-RUN npm prune --production
-WORKDIR /ui/ui-build
-RUN npm install cross-env
-CMD [ "/usr/bin/npm", "start", "--", "--port=8888", "--config=k8sdev", "--prefix", "/ui/ui-build" ]
 
+# run stage
+FROM node:12-alpine as run-stage
+
+# Change working directory
+RUN mkdir /ui
+WORKDIR /ui
+
+# Bring over app files
+COPY --from=build-stage /ui/build ./build
+COPY --from=build-stage /ui/config ./config
+COPY --from=build-stage /ui/dist ./dist
+COPY --from=build-stage /ui/server ./server
+COPY --from=build-stage /ui/src ./src
+COPY --from=build-stage /ui/package.json .
+COPY --from=build-stage /ui/package-lock.json .
+
+# Install PROD dependencies only
+RUN npm install --production
```

Comment:
So I think this can work for now, but if we want to slim down the final image, we should really examine what is needed at container runtime for the app. So *after* npm install, what is really needed to run the app? And then that set of assets is what we'd want to copy in.

The answer may just be "everything copied over above"... in which case, there may not be a lot of savings here.

However, we may still be able to optimize builds a bit by reordering. So e.g. if we could run npm install first, then copy over source files and run npm build (might not work that way... I fully admit I don't know anything about the UI app build process), then we might be able to leverage local docker image layer caching to speed up builds.

All this is just contextualizing some of the conversation we've had elsewhere closer to the Dockerfile, and I think could just be things to consider in future revs rather than needing to be addressed right now.

Reply:
Also agree. Just whittled these down with a rough pass. We may be able to drop the `src` folder but I'll to run some local tests with the app to determine exactly what is needed for a deployed instance.